File: app/src/main/java/org/ametro/ui/navigation/entities/NavigationSubHeader.java
package org.ametro.ui.navigation.entities;

public class NavigationSubHeader extends NavigationItem implements INavigationItemGroup {

    private final CharSequence text;
    private final NavigationItem[] items;

    public NavigationSubHeader(CharSequence text, NavigationItem[] items){
        this(INVALID_ACTION, text, items);
    }

    public NavigationSubHeader(int action, CharSequence text, NavigationItem[] items){
        super(action);
        this.text = text;
        this.items = items;
    }

    public CharSequence getText() {
        return text;
    }

    public NavigationItem[] getItems() {
        return items;
    }
}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/navigation/entities/NavigationSubHeader.kt
package io.github.romangolovanov.apps.ametro.ui.navigation.entities

class NavigationSubHeader(
    action: Int = INVALID_ACTION,
    val text: CharSequence,
    private val items: Array<NavigationItem>
) : NavigationItem(action), INavigationItemGroup {

    constructor(text: CharSequence, items: Array<NavigationItem>) : this(INVALID_ACTION, text, items)

    override fun getItems(): Array<NavigationItem> = items
}
